File: packages/web-app/src/models/Sprint.ts
export enum SprintStatus {
    active = 'active',
    closed = 'closed',
}

export interface Sprint {
    id: number;
    title: string;
    description: string;
    status: SprintStatus;
    startDate?: string;
    dueDate?: string;
    totalStoryPoint?: number;
    totalNumberOfIssues?: number;
    projectIdToMilestoneIds: { [key: string]: number };
}

export function isSprint(result: Sprint | string): result is Sprint {
    return !!(result as Sprint).id;
}

export function isSprintArray(result: Sprint[] | string): result is Sprint[] {
    return Array.isArray(result);
}

File: packages/lts/src/main/java/com/bt/scrumbleLts/dto/Sprint.java
package com.bt.scrumbleLts.dto;

import java.util.Map;

public class Sprint {
  private final int id;
  private final String title;
  private final String description;
  private final String status;
  private final String startDate;
  private final String dueDate;
  private final Map<String, Integer> projectIdToMilestoneIds;

  public Sprint(
      int id,
      String title,
      String description,
      String status,
      String startDate,
      String dueDate,
      Map<String, Integer> projectIdToMilestoneIds) {
    this.id = id;
    this.title = title;
    this.description = description;
    this.status = status;
    this.startDate = startDate;
    this.dueDate = dueDate;
    this.projectIdToMilestoneIds = projectIdToMilestoneIds;
  }
}
